From 9320adbea46f051b32c93a171dc0d52de8f845c3 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Fri, 14 Jun 2019 12:40:56 -0700
Subject: [PATCH] rsync: Add missing includes

- stdio.h in socket.c for sscanf
- stdint.h in extern.h for fixed-width integer types
- sys/types.h in extern.h for various type definitions
---
 usr.bin/rsync/extern.h | 2 ++
 usr.bin/rsync/socket.c | 1 +
 2 files changed, 3 insertions(+)

diff --git a/usr.bin/rsync/extern.h b/usr.bin/rsync/extern.h
index 5bb23aaa486..55d7bc26ae2 100644
--- a/usr.bin/rsync/extern.h
+++ b/usr.bin/rsync/extern.h
@@ -17,6 +17,8 @@
 #ifndef EXTERN_H
 #define EXTERN_H
 
+#include <stdint.h>
+#include <sys/types.h>
 #include <openssl/md4.h>
 
 /*
diff --git a/usr.bin/rsync/socket.c b/usr.bin/rsync/socket.c
index d723d9e7df4..adc4a22cf40 100644
--- a/usr.bin/rsync/socket.c
+++ b/usr.bin/rsync/socket.c
@@ -28,6 +28,7 @@
 #include <poll.h>
 #include <resolv.h>
 #include <stdlib.h>
+#include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 #include <err.h>
-- 
2.49.0

